web3.js/src/epoch-schedule.ts

```diff
@@ -40,6 +53,29 @@ export class EpochSchedule {
     this.firstNormalSlot = firstNormalSlot;
   }
 
+  getEpoch(slot: number): number {
+    return this.getEpochAndSlotIndex(slot)[0];
+  }
+
+  getEpochAndSlotIndex(slot: number): [number, number] {
+    if (slot < this.firstNormalSlot) {
+      const epoch =
+        trailingZeros(nextPowerOfTwo(slot + MINIMUM_SLOT_PER_EPOCH + 1)) -
+        trailingZeros(MINIMUM_SLOT_PER_EPOCH) -
+        1;
+
+      const epochLen = this.getSlotsInEpoch(epoch);
+      const slotIndex = slot - (epochLen - MINIMUM_SLOT_PER_EPOCH);
+      return [epoch, slotIndex];
+    } else {
+      const normalSlotIndex = slot - this.firstNormalSlot;
+      const normalEpochIndex = (normalSlotIndex / this.slotsPerEpoch) | 0;
```

Comment:
Nice, I didn't know about this truncation method! I would prefer if an explicit `Math.floor` is used though since `number | 0` only works for 32 bit integers (source: https://stackoverflow.com/a/7488075)